packages/eslint-plugin-nx/src/rules/enforce-module-boundaries.ts

```diff
@@ -388,24 +392,46 @@ export default createESLintRule<Options, MessageIds>({
 
         for (let constraint of constraints) {
           if (
+            constraint.onlyDependOnLibsWithTags &&
             hasNoneOfTheseTags(
               targetProject,
-              constraint.onlyDependOnLibsWithTags || []
+              constraint.onlyDependOnLibsWithTags
             )
           ) {
             const allowedTags = constraint.onlyDependOnLibsWithTags
               .map((s) => `"${s}"`)
               .join(', ');
             context.report({
               node,
-              messageId: 'tagConstraintViolation',
+              messageId: 'onlyTagsConstraintViolation',
               data: {
                 sourceTag: constraint.sourceTag,
                 allowedTags,
               },
             });
             return;
           }
+          if (
+            constraint.notDependOnLibsWithTags &&
+            hasAnyOfTheseTags(
+              projectGraph,
```

Comment:
Because I don't have a full ProjectGraphNode object within the [recursive call here](https://github.com/CurioVision/nx/blob/d0acaeea182622f5963538341bfaf73e2fed0a78/packages/workspace/src/utils/runtime-lint-utils.ts#L59-L60).

If this call goes away, we can and should change this to match the similar function signature on line 397. I think there's still some discussion to be had regarding this though.